src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java

```diff
@@ -163,24 +178,24 @@ public static Optional<String> getIndividualPreferredHearingChannel(String heari
             : Optional.empty();
     }
 
-    public static Optional<String> getIndividualInterpreterLanguage(HearingOptions hearingOptions) {
+    public static String getIndividualInterpreterLanguage(HearingOptions hearingOptions, ReferenceDataServiceHolder referenceData) throws InvalidMappingException {
         if (isTrue(hearingOptions.wantsSignLanguageInterpreter())) {
-            return getSignLanguage(hearingOptions)
-                .map(SignLanguage::getHmcReference);
+            String signLanguage = hearingOptions.getSignLanguageType();
+            String signLanguageReference = referenceData.getSignLanguages().getSignLanguageReference(signLanguage);
+            if (isNull(signLanguageReference)) {
+                throw new InvalidMappingException(String.format("The language %s cannot be mapped", signLanguage));
+            }
+            return signLanguageReference;
         }
         if (isYes(hearingOptions.getLanguageInterpreter())) {
-            return getInterpreterLanguage(hearingOptions)
-                .map(InterpreterLanguage::getHmcReference);
+            String verbalLanguage = hearingOptions.getLanguages();
+            String verbalLanguageReference = referenceData.getVerbalLanguages().getVerbalLanguageReference(verbalLanguage);
+            if (isNull(verbalLanguageReference)) {
+                throw new InvalidMappingException(String.format("The language %s cannot be mapped", verbalLanguage));
+            }
+            return verbalLanguageReference;
         }
-        return Optional.empty();
-    }
-
-    private static Optional<SignLanguage> getSignLanguage(HearingOptions hearingOptions) {
-        return Optional.ofNullable(SignLanguage.getSignLanguageKeyByCcdReference(hearingOptions.getSignLanguageType()));
-    }
-
-    private static Optional<InterpreterLanguage> getInterpreterLanguage(HearingOptions hearingOptions) {
-        return Optional.ofNullable(InterpreterLanguage.getLanguageAndConvert(hearingOptions.getLanguages()));
+        return null;
     }
```

Comment:
Optional: Does this need a TODO for later work?

Reply:
I don't like todo's for later work, otherwise we end up with the code riddled with todo's when we should always be refactoring as we develop, so better to only use todo's whilst you're putting your code together, then ensure they are all gone before raising the PR

Reply:
We talked this over, this is intentional